### ISSUE TITLE
Missing skip connection for cross attention?

### ISSUE THREAD

**Issue by lonzi** [lucidrains/audiolm-pytorch#150]
Hi, thanks for the implementation, it is really helpful!

While using the Transformer class, I noticed a significant performance degradation when activating the optional cross attention branch. I suspect this is due to a missing skip connection in the following line of code:
https://github.com/lucidrains/audiolm-pytorch/blob/main/audiolm_pytorch/audiolm_pytorch.py#L456

I suggest changing to 
x = cross_attn(x, context = context, mask = context_mask) **+ x**

**Comment by lucidrains** [comment]
@lonzi i was just trying to see if you were paying attenti.. just kidding

that was one of my high school teacher's favorite line :laughing:

thank you for catching this Alon!

**Comment by lonzi** [comment]
Thanks for fixing! No, I work on drums generation and using the mentioned transformer above a pre-trained EnCodec model.
